### PR TITLE
fix(locale-selector): changed icon and em-dash added

### PR DESCRIPTION
**Pull request by guilhermelMoraes**
### Related Ticket(s)

Locale selector, button/indicator using incorrect dash and icon [#1738](https://github.com/carbon-design-system/ibm-dotcom-library/issues/1738)

### Description

Changed the icon from `Globe20` to `EarthFilled20` and replaced the dash to em-dash in both indicator/buttons from locale selector.

### CODE REVIEW

File: packages/react/src/components/Footer/Footer.js [hunk 1]
```diff
@@ -58,7 +58,8 @@ const Footer = ({ type, navigation, langCode, disableLocaleButton }) => {
   useEffect(() => {
     (async () => {
       const response = await LocaleAPI.getLangDisplay(langCode);
-      setDisplayLang(response);
+      const formatedLangDisplay = response.replace('-', '—');
```
Comment (jeffchew) [inline packages/react/src/components/Footer/Footer.js:61]: This should actually be replaced in the LocaleAPI service itself. It is in the services package in the same repo.

File: packages/react/src/components/LocaleModal/LocaleModal.js [hunk 2]
```diff
@@ -44,7 +44,8 @@ const LocaleModal = ({ isOpen, setIsOpen }) => {
       const locale = await LocaleAPI.getLocale();
       const list = locale && (await LocaleAPI.getList(locale));
       const getLangDisplay = await LocaleAPI.getLangDisplay();
-      setLangDisplay(getLangDisplay);
+      const formatedLangDisplay = getLangDisplay.replace('-', '—');
```
Comment (jeffchew) [inline packages/react/src/components/LocaleModal/LocaleModal.js:47]: Same here, make this change in the service and not here.

**Comment by jeffchew** [comment]
@guilhermelMoraes make sure you update the unit test too:
https://app.circleci.com/pipelines/github/carbon-design-system/ibm-dotcom-library/1111/workflows/cee6dc77-8174-49fc-bf63-80d73e06256a/jobs/2090

Screen Shot 2020-04-14 at 2 18 19 PM

**Comment by jeffchew** [comment]
@guilhermelMoraes I have also added the `package: react` label so that we get the deploy previews for React.

**Comment by guilhermelMoraes** [comment]
> @guilhermelMoraes make sure you update the unit test too:
> https://app.circleci.com/pipelines/github/carbon-design-system/ibm-dotcom-library/1111/workflows/cee6dc77-8174-49fc-bf63-80d73e06256a/jobs/2090
> 
> Screen Shot 2020-04-14 at 2 18 19 PM

@jeffchew I can't figure it out why the checks keep failing. I've updated the unit test from the Locale service.

**Comment by jeffchew** [comment]
> @jeffchew I can't figure it out why the checks keep failing. I've updated the unit test from the Locale service.

This is your answer:

```
Expected: "United States - English"
Received: "United States — English"
```

You should update the test to expect a value with the em dash and not a regular dash:

https://github.com/carbon-design-system/ibm-dotcom-library/blob/534098a31eb3c330561183ebe01e63322e7f89aa/packages/services/src/services/Locale/__tests__/Locale.test.js#L73

**Comment by guilhermelMoraes** [comment]
> > @jeffchew I can't figure it out why the checks keep failing. I've updated the unit test from the Locale service.
> 
> This is your answer:
> 
> ```
> Expected: "United States - English"
> Received: "United States — English"
> ```
> 
> You should update the test to expect a value with the em dash and not a regular dash:
> 
> https://github.com/carbon-design-system/ibm-dotcom-library/blob/534098a31eb3c330561183ebe01e63322e7f89aa/packages/services/src/services/Locale/__tests__/Locale.test.js#L73

Strange. That's exactly what I've did. https://github.com/carbon-design-system/ibm-dotcom-library/blob/a066d3dc83f0bf31569fd64e036bcef81db532b2/packages/services/src/services/Locale/__tests__/Locale.test.js#L73

**Comment by annawen1** [comment]
@guilhermelMoraes maybe you need to update your snapshots as well?

try running `npm test -- -u` on the root of the project and push the changed snapshot files if there are any

**Pull request merged**